src/shared/parameter_v2/config/examples/generated_c_test.cpp

```diff
@@ -0,0 +1,61 @@
+#include <gtest/gtest.h>
+
+#include "shared/parameter_v2/config/examples/example_config.h"
+#include "shared/parameter_v2/config/examples/parameter.h"
+
+TEST(BasicAccess, test_basic_parameter_access)
+{
+    const ThunderbotsConfig_t* tbots_config = app_dynamic_parameters_create();
+
+    ASSERT_TRUE(tbots_config->ExampleConfig->FooConfig->foo_bool.value == true);
+    ASSERT_TRUE(tbots_config->ExampleConfig->BarConfig->bar_bool.value == true);
+    ASSERT_TRUE(tbots_config->ExampleConfig->FooConfig->foo_int.value == 3);
+    ASSERT_TRUE(tbots_config->ExampleConfig->BarConfig->bar_int.value == 3);
+
+    // WONT COMPILE
+    // tbots_config->ExampleConfig->BarConfig->bar_bool.value = true;
+    // tbots_config->ExampleConfig->FooConfig->foo_int.value = 3;
```

Comment:
I definitely see why we have this here, just not sure if we should keep it or not. Generally we don't put code snippets that don't compile in tests.

@MathewMacDougall @jonathanlew @EvanMorcom  - your thoughts?